// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: event/v1/service.proto

package v1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on IngestEventRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *IngestEventRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on IngestEventRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// IngestEventRequestMultiError, or nil if none found.
func (m *IngestEventRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *IngestEventRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return IngestEventRequestMultiError(errors)
	}

	return nil
}

// IngestEventRequestMultiError is an error wrapping multiple validation errors
// returned by IngestEventRequest.ValidateAll() if the designated constraints
// aren't met.
type IngestEventRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m IngestEventRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m IngestEventRequestMultiError) AllErrors() []error { return m }

// IngestEventRequestValidationError is the validation error returned by
// IngestEventRequest.Validate if the designated constraints aren't met.
type IngestEventRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e IngestEventRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e IngestEventRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e IngestEventRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e IngestEventRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e IngestEventRequestValidationError) ErrorName() string {
	return "IngestEventRequestValidationError"
}

// Error satisfies the builtin error interface
func (e IngestEventRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sIngestEventRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = IngestEventRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = IngestEventRequestValidationError{}

// Validate checks the field values on IngestEventResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *IngestEventResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on IngestEventResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// IngestEventResponseMultiError, or nil if none found.
func (m *IngestEventResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *IngestEventResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Success

	for idx, item := range m.GetErrors() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, IngestEventResponseValidationError{
						field:  fmt.Sprintf("Errors[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, IngestEventResponseValidationError{
						field:  fmt.Sprintf("Errors[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return IngestEventResponseValidationError{
					field:  fmt.Sprintf("Errors[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return IngestEventResponseMultiError(errors)
	}

	return nil
}

// IngestEventResponseMultiError is an error wrapping multiple validation
// errors returned by IngestEventResponse.ValidateAll() if the designated
// constraints aren't met.
type IngestEventResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m IngestEventResponseMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m IngestEventResponseMultiError) AllErrors() []error { return m }

// IngestEventResponseValidationError is the validation error returned by
// IngestEventResponse.Validate if the designated constraints aren't met.
type IngestEventResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e IngestEventResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e IngestEventResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e IngestEventResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e IngestEventResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e IngestEventResponseValidationError) ErrorName() string {
	return "IngestEventResponseValidationError"
}

// Error satisfies the builtin error interface
func (e IngestEventResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sIngestEventResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = IngestEventResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = IngestEventResponseValidationError{}

// Validate checks the field values on IngestEventResponse_ErrorWrapper with
// the rules defined in the proto definition for this message. If any rules
// are violated, the first error encountered is returned, or nil if there are
// no violations.
func (m *IngestEventResponse_ErrorWrapper) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on IngestEventResponse_ErrorWrapper with
// the rules defined in the proto definition for this message. If any rules
// are violated, the result is a list of violation errors wrapped in
// IngestEventResponse_ErrorWrapperMultiError, or nil if none found.
func (m *IngestEventResponse_ErrorWrapper) ValidateAll() error {
	return m.validate(true)
}

func (m *IngestEventResponse_ErrorWrapper) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for Error

	if len(errors) > 0 {
		return IngestEventResponse_ErrorWrapperMultiError(errors)
	}

	return nil
}

// IngestEventResponse_ErrorWrapperMultiError is an error wrapping multiple
// validation errors returned by
// IngestEventResponse_ErrorWrapper.ValidateAll() if the designated
// constraints aren't met.
type IngestEventResponse_ErrorWrapperMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m IngestEventResponse_ErrorWrapperMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m IngestEventResponse_ErrorWrapperMultiError) AllErrors() []error { return m }

// IngestEventResponse_ErrorWrapperValidationError is the validation error
// returned by IngestEventResponse_ErrorWrapper.Validate if the designated
// constraints aren't met.
type IngestEventResponse_ErrorWrapperValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e IngestEventResponse_ErrorWrapperValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e IngestEventResponse_ErrorWrapperValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e IngestEventResponse_ErrorWrapperValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e IngestEventResponse_ErrorWrapperValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e IngestEventResponse_ErrorWrapperValidationError) ErrorName() string {
	return "IngestEventResponse_ErrorWrapperValidationError"
}

// Error satisfies the builtin error interface
func (e IngestEventResponse_ErrorWrapperValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sIngestEventResponse_ErrorWrapper.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = IngestEventResponse_ErrorWrapperValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = IngestEventResponse_ErrorWrapperValidationError{}
